src/main/java/wooteco/subway/path/domain/SubwayPath.java

```diff
@@ -1,16 +1,38 @@
 package wooteco.subway.path.domain;
 
+import wooteco.subway.line.domain.Line;
+import wooteco.subway.path.strategy.FareStrategy;
 import wooteco.subway.station.domain.Station;
 
 import java.util.List;
+import java.util.Set;
 
 public class SubwayPath {
-    private List<SectionEdge> sectionEdges;
-    private List<Station> stations;
+    private final List<SectionEdge> sectionEdges;
+    private final List<Station> stations;
+    private final Set<Line> lines;
+    private final FareStrategy fareStrategy;
 
-    public SubwayPath(List<SectionEdge> sectionEdges, List<Station> stations) {
+    public SubwayPath(List<SectionEdge> sectionEdges, List<Station> stations, Set<Line> lines, FareStrategy fareStrategy) {
         this.sectionEdges = sectionEdges;
         this.stations = stations;
+        this.lines = lines;
+        this.fareStrategy = fareStrategy;
+    }
+
+    public int calculateFare() {
+        return fareStrategy.discount(calculateDistance(), findExtraFare());
+    }
+
+    private int findExtraFare() {
+        return lines.stream()
+                .mapToInt(Line::getExtraFare)
+                .max()
+                .orElse(0);
+    }
+
+    public int calculateDistance() {
+        return sectionEdges.stream().mapToInt(it -> it.getSection().getDistance()).sum();
```

Comment:
```suggestion
        return sectionEdges.stream()
                .mapToInt(SectionEdge::getSectionDistance)
                .sum();
```

- stream 가공은 가독성을 위해 한줄에 한 단계씩 작성해 주세요 ㅎㅎ
- 디미터 법칙을 적용해 보아요 :)